pkg/apis/serving/k8s_lifecycle_test.go

```diff
@@ -158,12 +160,45 @@ func TestTransformDeploymentStatus(t *testing.T) {
 				Type:    DeploymentConditionReplicaSetReady,
 				Status:  corev1.ConditionFalse,
 				Reason:  "ReplicaSetReason",
-				Message: "Something bag happened",
+				Message: "Something bad happened",
 			}, {
 				Type:    DeploymentConditionReady,
 				Status:  corev1.ConditionFalse,
 				Reason:  "ReplicaSetReason",
-				Message: "Something bag happened",
+				Message: "Something bad happened",
+			}},
+		},
+	}, {
+		name: "replica failure has priority over progressing",
```

Comment:
Could you elaborate where priority is defined, I see that `DeploymentConditionProgressing` is the same before and after, so no change there.

Reply:
'priority' here means that the replicafailure message is the last one applied so it is surfaced to the deployment's Ready condition.